lib/handler/configurator/headers_util.c

```diff
@@ -45,18 +45,34 @@ static int extract_name_value(const char *src, h2o_iovec_t **name, h2o_iovec_t *
     return 0;
 }
 
-static int add_cmd(h2o_configurator_command_t *cmd, yoml_t *node, int cmd_id, h2o_iovec_t *name, h2o_iovec_t value,
-                   h2o_headers_command_when_t when, h2o_headers_command_t **cmds)
+static int is_list_cmd(int cmd_id)
 {
-    if (h2o_iovec_is_token(name)) {
-        const h2o_token_t *token = (void *)name;
-        if (h2o_headers_is_prohibited_name(token)) {
-            h2o_configurator_errprintf(cmd, node, "the named header cannot be rewritten");
-            return -1;
+    return cmd_id == H2O_HEADERS_CMD_UNSET
+        || cmd_id == H2O_HEADERS_CMD_UNSETUNLESS
+        || cmd_id == H2O_HEADERS_CMD_COOKIE_UNSET
+        || cmd_id == H2O_HEADERS_CMD_COOKIE_UNSETUNLESS;
+}
+
+static int add_cmd(h2o_configurator_command_t *cmd, yoml_t **node, int cmd_id, h2o_iovec_t **name, h2o_iovec_t *value,
```

Comment:
How about creating a struct for carrying (node, name, value), and changing the function definition to something like:

```c
struct st_h2o_headers_add_arg_t {
    yoml_t *node;
    h2o_iovec_t *name;
    h2o_iovec_t value;
};
static int ad_command(h2o_configurator_command_t *cmd, int cmd_id, struct st_h2o_headers_add_arg_t *args, size_t num_args, h2o_headers_command_when_t when, h2o_headers_command_t **cmds)
{
...
```